Update the documentation regarding ICU (#1332)
freedomofpress/dangerzone: docs/developer/release/build.md

# Build release artifacts

Follow the sections below, depending on the OS you want to build artifacts for.

> Linux artifacts can be built from macOS, via the use of containers.

Reviewer: ```suggestion
Linux artifacts can be built on macOS using containers.
```

Author: Thanks. I applied one half of your suggestion, as I think [the "containers" project from macOS](https://github.com/apple/container) might confuse some people otherwise.
